core/datastore/src/main/java/com/google/samples/apps/nowinandroid/core/datastore/NiaPreferencesDataSource.kt

```diff
@@ -188,4 +193,19 @@ class NiaPreferencesDataSource @Inject constructor(
             Log.e("NiaPreferences", "Failed to update user preferences", ioException)
         }
     }
+
+    suspend fun setShouldHideOnboarding(shouldHideOnboarding: Boolean) {
+        userPreferences.updateData {
+            it.copy {
+                this.shouldHideOnboarding = shouldHideOnboarding
+            }
+        }
+    }
+}
+
+fun UserPreferencesKt.Dsl.updateShouldHideOnboardingIfNecessary() {
```

Comment:
Minor, this should probably be private since it should only be called atomically.